### ISSUE TITLE
VSCode get frozen when editing a particular React code snippets

### ISSUE THREAD

**Issue by jesseminn** [vincaslt/vscode-highlight-matching-tag#35]
I'm using this extension with VSCode VIM, VSCode get frozen when the cursor moved  inside of the JSX part of this code snippets:

```tsx
class Game extends React.PureComponent<IProps> {
    render(): JSX.Element {
        const { isMobile }: IProps = this.props;
        return (
            <Carousel
                dots={!isMobile}
                slidesToShow={isMobile ? 1.5 : 3}
                slidesToScroll={isMobile ? 1.5 : 3}
                dotsClass="game-dots"
                arrows={!isMobile}
                prevArrow={<div>{'<'}</div>}
                nextArrow={<div>{'>'}</div>}
            >
            </Carousel>
        );
    }
}
```

Tested in VSCode Insiders, with only VSCode VIM & Highlight Matching Tag installed, no extra settings applied.

After disabling or uninstalling HMT VSCode won't get frozen, thus I think HMT causes this issue.

Interestingly, if I did one of the following adjustment, VSCode won't get frozen:

- remove the curly braces `{...}` inside of `prevArrow` prop.
- remove any prop before `prevArrow` prop.
- or Disabling VSCode VIM

I know it's a very weird issue, but I hope it make sense.

**Comment by lululombard** [comment]
I have the exact same issue. Had to disable it for my reactjs projects.

**Comment by vincaslt** [comment]
Should be fixed with v0.5, test cases cover it now.
